dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestInjectorInterface.cpp

```diff
@@ -0,0 +1,124 @@
+//
+// Copyright (c) 2015-2020, RTE (http://www.rte-france.com)
+// See AUTHORS.txt
+// All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+// This file is part of Dynawo, an hybrid C++/Modelica open source time domain
+// simulation tool for power systems.
+//
+
+#include "DYNInjectorInterfaceIIDM.h"
+
+#include "DYNBusInterfaceIIDM.h"
+#include "DYNLoadInterfaceIIDM.h"
+#include "DYNVoltageLevelInterfaceIIDM.h"
+
+#include <powsybl/iidm/Bus.hpp>
+#include <powsybl/iidm/Load.hpp>
+#include <powsybl/iidm/LoadAdder.hpp>
+#include <powsybl/iidm/Network.hpp>
+#include <powsybl/iidm/Substation.hpp>
+
+#include "gtest_dynawo.h"
+
+namespace DYN {
+
+TEST(DataInterfaceTest, Injector) {
+  using powsybl::iidm::Bus;
+  using powsybl::iidm::Load;
+  using powsybl::iidm::LoadType;
+  using powsybl::iidm::Network;
+  using powsybl::iidm::Substation;
+  using powsybl::iidm::TopologyKind;
+  using powsybl::iidm::VoltageLevel;
+
+  Network network("test", "test");
+  Substation& substation = network.newSubstation()
+                               .setId("S1")
+                               .setName("S1_NAME")
+                               .setCountry(powsybl::iidm::Country::FR)
+                               .setTso("TSO")
+                               .add();
+
+  VoltageLevel& vl1 = substation.newVoltageLevel()
+                          .setId("VL1")
+                          .setName("VL1_NAME")
+                          .setTopologyKind(TopologyKind::BUS_BREAKER)
+                          .setNominalVoltage(380.0)
+                          .setLowVoltageLimit(340.0)
+                          .setHighVoltageLimit(420.0)
+                          .add();
+
+  Bus& bus1 = vl1.getBusBreakerView().newBus().setId("VL1_BUS1").add();
+
+  vl1.newLoad()
+      .setId("LOAD1")
+      .setBus("VL1_BUS1")
+      .setConnectableBus("VL1_BUS1")
+      .setName("LOAD1_NAME")
+      .setLoadType(LoadType::UNDEFINED)
+      .setP0(50.0)
+      .setQ0(40.0)
+      .add();
+
+  substation.newVoltageLevel()
+      .setId("VL2")
+      .setName("VL2_NAME")
+      .setTopologyKind(TopologyKind::NODE_BREAKER)
+      .setNominalVoltage(225.0)
+      .setLowVoltageLimit(200.0)
+      .setHighVoltageLimit(260.0)
+      .add();
+
+  Load& load = network.getLoad("LOAD1");
+  ASSERT_EQ("LOAD1", load.getId());
```

Comment:
This is the load test, not the injection one. Please rename the file accordingly

Reply:
I cannot test Injector without another class. I confirm it is Injector's test through "Load". We can test with other classes having an Injector, but there must be a chosen one. I may cut lines only about Load, or make distinct lines specially for injector. But both now are built together. May I keep it until the next class built with an Injector ? Later we may suppress that file (it does not exist as a model, it is tested through other classes)

Reply:
Injector will be tested through the child classes. In this test we test many things related to Load and not Injector, so let's say this one is the Load test and rename the file

Reply:
Finaly I switched to Load to cntinue. Load tests InjectorInterface and LoadInterface :)